clippy_lints/src/if_not_else.rs

```diff
@@ -61,8 +61,12 @@ impl EarlyLintPass for IfNotElse {
         if in_external_macro(cx.sess(), item.span) {
             return;
         }
-        if let ExprKind::If(ref cond, _, Some(ref els)) = item.node {
-            if let ExprKind::Block(..) = els.node {
+        if let ExprKind::If(ref cond, ref thn, Some(ref els)) = item.node {
+            if let ExprKind::Block(ref els_block, ..) = els.node {
+                if thn.stmts.len() < els_block.stmts.len() {
```

Comment:
we could make this lint configurable over a ratio. I think something like 1:2 would still be ok as a default

Reply:
True, can you open an issue for adding such a ratio to our config?

Reply:
Will do.

Unrelated, but could you take a look at https://github.com/Wilfred/remacs/pull/937/commits/dee6781384d2472235539142af4e51a0534a3596 and see if it takes care of https://github.com/Wilfred/remacs/issues/757?

Reply:
Number of lines is also something to consider. For instance
```rust
                if self.q != 0 {
                    if self.tail == self.tortoise {
                        return self.circular();
                    }
                } else {
                    self.n = self.n.wrapping_sub(1);
                    if self.n > 0 {
                        if self.tail == self.tortoise {
                            return self.circular();
                        }
                    } else {
                        self.max <<= 1;
                        self.q = self.max as u16;
                        self.n = self.max >> 16;
                        self.tortoise = self.tail;
                    }
```
There are just two statements in the `else` block (assuming the `if` counts as one), but the whole block is pretty long, and I wouldn't want to switch those.